nomad/encrypter.go

```diff
@@ -435,7 +456,10 @@ START:
 			return
 		default:
 			// Rate limit how often we attempt replication
-			limiter.Wait(ctx)
+			err := limiter.Wait(ctx)
+			if err != nil {
+				goto ERR_WAIT // rate limit exceeded
+			}
```

Comment:
Seems like if we have our own blocking mechanism, we should be using `Allow` instead? Otherwise we are kind of double-waiting, which is fine but awkward given the error. (no need to change anything now)

https://pkg.go.dev/golang.org/x/time/rate#Limiter

Reply:
Yeah this API is weirdly hard to use correctly and the error case is _only_ for exceeding the burst. I first thought we probably wanted to use `Reserve` but that has the same double-waiting problem. Maybe we should just remove the burst (set it to `Inf`) so that we guarantee we'll wait and not have to handle this error so awkwardly?

Reply:
I misread the API again 😊 . Only when the _rate_ is set to `rate.Inf` is the burst-limit ignored. So this is good as-is, awkward as it seems.